bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/Action.java

```diff
@@ -26,7 +27,22 @@
  * @author Ana Dimova - Initial Contribution
  * @author Vasil Ilchev - Initial Contribution
  */
-public interface Action extends Module {
+public class Action extends Module {
+
+    protected Set<Connection> connections;
+
+    /**
+     * Constructor of Action object.
+     *
+     * @param UID action unique id.
+     * @param typeUID module type unique id.
+     * @param configuration map of configuration values.
+     * @param connections set of connections to other modules (triggers and other actions).
+     */
+    public Action(String UID, String typeUID, Map<String, ?> configuration, Set<Connection> connections) {
+        super(UID, typeUID, configuration);
+        setConnections(connections);
```

Comment:
I think it is always better to use a setter then just variable assignment. The setter is needed because if we want to change connections of action it is not needed  to create new Action object.